### DISCUSSION TITLE
Authorization services not work

### DISCUSSION THREAD

**Discussion by MorganReid** [keycloak/keycloak#20779] [Q&A]
I want to implement a client to only open permissions to specified users, and complete the configuration through [official documents](https://www.keycloak.org/docs/latest/authorization_services/index.html#providing-identity-information), but it does not take effect.
`client capability config:`
![image](https://github.com/keycloak/keycloak/assets/52235796/2b70e363-e0ae-463e-92fe-cdf09054ca5f)
`client Authorization Resource:`
![image](https://github.com/keycloak/keycloak/assets/52235796/0f5e8584-d4d3-4e02-b478-876791b431f8)
`client Authorization policy`:Create a new policy named `mypolicy` whose `Type` is `User`, and specify user `wiki1`
![image](https://github.com/keycloak/keycloak/assets/52235796/5a9eecbb-f1f4-4ec4-9f53-2b66dee09e82)
`client Authorization permission`:Create a new permission whose `Type` is `Resource-Based`, and specify policy `mypolicy`,specific resources  `Default Resource`
![image](https://github.com/keycloak/keycloak/assets/52235796/c63e44e3-b644-4230-b21c-240dda471334)
when i use `Evaluate` test,looks as expected
use wiki1:
![image](https://github.com/keycloak/keycloak/assets/52235796/053c1e44-6ec7-44e6-a8d8-38e35463bf6d)
use wiki2:
![image](https://github.com/keycloak/keycloak/assets/52235796/d8e157b9-0f7f-48a3-a9da-a603f9a49733)
But when I access my client(`OpenID Connect`), it will use keycloak for login authentication. Both wiki1 and wiki2 can be authenticated.Is there any configuration I'm missing? hope you can help me.

**Comment by ByteAfterlife** [comment]
Same issue for me

**Reply by v-miguel** [reply]
same here ...

**Comment by ByteAfterlife** [comment]
Ok I think I solved the issue, in keycloak, when authorization is being done, keycloak doesn't stop the user from signing in ever, instead what it does is in the OpenID Connect token being given, it includes a new claim about which resources the user should be allowed to access, this is up to the application to interpret, not keycloak. If your application does not understand how to restrict certain users from logging in based on authorization, you can create a custom flow that checks for a claim and then denies access at the keycloak level. That would however require some more additional steps. Tell me if you want me to tell you how to do that.

**Reply by trololirko** [reply]
I have the same issue. I'm very interested in knowing how you created a custom flow so that it checks for a claim, as I could not find this option in keycloaks authentication flows.

**Reply by souravs17031999** [reply]
What your requirement is actually a policy enforcement engine which needs to be always implemented on client side, keycloak can tell if the access to the resource is permitted or denied but it depends on your usecase what you want to achieve.
Also, keycloak does provide some capabilities for policy enforcement at client side for few stacks (java AFAIK), maybe there are more need tl check but for now, check this out for reference: 
https://www.keycloak.org/securing-apps/policy-enforcer
